### PR TITLE
making shutdown happen via channel+refactoring

### PR DESCRIPTION
**Pull request by r-medina**
- move constant defs out of function calls
- change timeouts
- add dialoptions to everything
- last golint comments

### CODE REVIEW

File: finger_test.go [hunk 1]
```diff
@@ -83,7 +83,7 @@ func TestStabilizedFingerTable(t *testing.T) {
 	node1, node2, node3 := create3SuccessiveNodes(t)
 
 	// Should be enough time to stabilize finger tables.
-	<-time.After(time.Second)
+	<-time.After(1500 * time.Millisecond)
```
Comment (asubiotto) [inline finger_test.go]: Why this? I would make a const

File: kv_store_test.go [hunk 2]
```diff
@@ -158,9 +158,9 @@ func TestTransferKeysAvailability(t *testing.T) {
 	// Make node that should get the key transferred to it.
 	hashedKey = HashKey(key)
 	hashedKey[0]++
-	node2, _ := NewDefinedNode(&node1.remoteNode, hashedKey)
+	node2, _ := NewDefinedNode(node1.RemoteNode(), hashedKey)
 
-	<-time.After(50 * time.Millisecond)
+	<-time.After(666 * time.Millisecond)
```
Comment (asubiotto) [inline kv_store_test.go]: same

**Comment by r-medina** [comment]
it was taking longer for things to converge, so i had to change those values. i'll make them consts in another pr if that's cool with you

**Comment by asubiotto** [comment]
No problem. It makes more sense to me to have the additional consts added in this PR as it seems to be the same logical unit of work

**Pull request closed without merging**